test/distributed/_tensor/test_matrix_ops.py

```diff
@@ -39,6 +39,25 @@ def test_addmm(self):
         local_res = torch.addmm(input_tensor, tensor_to_shard, tensor_to_replicate)
         self.assertEqual(dist_res.full_tensor(), local_res)
 
+    @with_comms
+    def test_addmm_broadcast(self):
+        device_mesh = DeviceMesh(self.device_type, list(range(self.world_size)))
+        shard_spec = [Shard(0)]
+        replica_spec = [Replicate()]
+
+        tensor_to_shard = torch.randn(12, 0)
+        mat1 = distribute_tensor(tensor_to_shard, device_mesh, shard_spec)
+        tensor_to_replicate = torch.randn(0, 4)
+        mat2 = distribute_tensor(tensor_to_replicate, device_mesh, replica_spec)
+        input_tensor = torch.randn(4)
+        input = distribute_tensor(input_tensor, device_mesh, replica_spec)
+
+        dist_res = torch.addmm(input, mat1, mat2)
+        local_res = torch.addmm(input_tensor, tensor_to_shard, tensor_to_replicate)
+        print(f"rank={self.rank}, dist_res._spec={dist_res._spec}, dist_res._local_tensor={dist_res._local_tensor}")
+        print(f"rank={self.rank}, dist_res={dist_res.full_tensor()}, local_res={local_res}")
+        self.assertNotEqual(dist_res.full_tensor(), local_res)
```

Comment:
I'm not following this logic, after your fix shouldn't the distributed tensor match the local tensor?

Reply:
It should be, so I mark this PR as WIP :-(

Unfortunately I haven't found a good solution on this issue. Initially I thought Wanchao's PR #115525 fixed this case but it turned out to be a NO...

I will leave that to another PR since it's caused by another issue in DTensor.